### PR TITLE
feat: HTTP proxy support

### PR DESCRIPTION
**Pull request by markmehere**
## Describe Your Changes

This adds requested proxy support although not SOCKS nor PAC support.

If you can decipher those acronyms you may know more about proxies than me.

It includes all changes from #1558 and the two work well in combination as many proxies invalidate end-to-end security.

Self-signed/unverified certificate support is understandably disabled by default - although on a certificate error the user will be notified of the setting.

Proxies must begin with `http` at all-levels to work or be saved to preferences.

To test proxy support:

```sh
brew install mitmproxy
mitmproxy -p 9028 --set stream_large_bodies=1m    # stream_large_bodies very important
make dev
# set HTTPS proxy to http://127.0.0.1:9028
# enable Ignore SSL certificates
# download a model and observe
```

Feel free to merge and edit - as it's your project! 🙂

## Screenshot

proxy-working

## Fixes Issues

- #1512 
- #1074 
- #1040 
- #1558 

## Self Checklist

- [x] Added relevant comments, esp in complex areas
- [x] Updated docs (for bug fixes / features)
- ~Created issues for follow-up changes or refactoring needed~

### CODE REVIEW

**Comment by 0xSage** [comment]
Wow thank you so much for this PR! We'll take a look latest Monday. 🙏 ❤️

**Comment by namchuai** [comment]
Overally looks good to me. Please put your comment @louis-jan

File: web/context/FeatureToggle.tsx [hunk 1]
```diff
@@ -18,25 +26,49 @@ export default function FeatureToggleWrapper({
...
-    setExperimentalEnabled(
-      localStorage.getItem(EXPERIMENTAL_FEATURE_ENABLED) === 'true'
+    directSetExperimentalFeature(
+      localStorage.getItem(EXPERIMENTAL_FEATURE) === 'true'
+    )
+    directSetIgnoreSSL(
+      localStorage.getItem(IGNORE_SSL) === 'true'
+    )
+    directSetProxy(
+      localStorage.getItem(HTTPS_PROXY_FEATURE)
```
Comment (louis-jan) [inline web/context/FeatureToggle.tsx:44]: Proxy prefers not to be null; localStorage.getItem(HTTPS_PROXY_FEATURE) might return null

**Review by louis-jan — CHANGES_REQUESTED** [review]
That's an awesome PR.

Please help resolve the error and rebase dev again.

Thanks

File: web/context/FeatureToggle.tsx (hunk 1 above)
Comment (louis-jan) [inline web/context/FeatureToggle.tsx:44]: @markmehere, that's a really great PR. I'd like to apply this change to fix the compilation error, and then we'll be good to go.
```suggestion
      localStorage.getItem(HTTPS_PROXY_FEATURE) ?? ""
```

**Review by hiento09 — APPROVED** [review]
LGTM, thank you so much, @markmehere, for dedicating your time to contributing to our repo.

**Review by louis-jan — APPROVED** [review]

**Comment by markmehere** [comment]
Cheers guys. Looks good. I had some build troubles - still trying to figure out why builds work on one machine and not the other. 😁

**Comment by Van-QA** [comment]
tested and looking good via the testing on Jan v0.4.4-160
image

**Pull request merged**